src/lib/components/events/event-carousel.svelte

```diff
@@ -1,108 +1,15 @@
 <script lang="ts">
-  import { onMount } from 'svelte';
   import EventItem from './event-item.svelte';
   import type { AcmEvent } from '$lib/ical/parse';
-  import { Time } from '$lib/constants/time';
 
   export let events: AcmEvent[] = [];
-
-  const scrollIncrementDistance = 335;
-  let carouselRef: HTMLDivElement;
-  let carouselButtonLeft: HTMLDivElement;
-  let carouselButtonRight: HTMLDivElement;
-  let isGrabbing = false;
-  let hasHorizontalScrollbar = false;
-  let leftButtonEnabled = false;
-  let rightButtonEnabled = false;
-  let scrollTimeId = undefined;
-
-  function scrollTheCarousel(movementScalar: number, isSmooth = false) {
-    carouselRef.scrollBy({
-      left: -movementScalar,
-      behavior: isSmooth ? 'smooth' : 'auto',
-    });
-    debounceTheScroll(() => {
-      const canScrollLeft = carouselRef.scrollLeft > 0;
-      const canScrollRight =
-        carouselRef.scrollWidth - carouselRef.scrollLeft - carouselRef.clientWidth > 1;
-      leftButtonEnabled = canScrollLeft;
-      rightButtonEnabled = canScrollRight;
-    }, Time.Second * 0.75);
-  }
-
-  function debounceTheScroll(callback: () => void, wait: number) {
-    callback();
-    if (scrollTimeId !== undefined) {
-      clearTimeout(scrollTimeId);
-    }
-
-    scrollTimeId = setTimeout(callback, wait);
-  }
-
-  function scrollOnMouseMove(event: MouseEvent) {
-    if (isGrabbing) {
-      scrollTheCarousel(event.movementX);
-    }
-  }
-
-  function startGrabbing() {
-    isGrabbing = true;
-  }
-
-  function endGrabbing() {
-    isGrabbing = false;
-  }
-
-  function scrollLeft() {
-    scrollTheCarousel(scrollIncrementDistance, true);
-  }
-
-  function scrollRight() {
-    scrollTheCarousel(-scrollIncrementDistance, true);
-  }
-
-  function scrollHorizontally(event: WheelEvent) {
-    event.preventDefault();
-    scrollTheCarousel(-event.deltaY);
-  }
-
-  onMount(() => {
-    hasHorizontalScrollbar = carouselRef.scrollWidth > carouselRef.clientWidth;
-    rightButtonEnabled = hasHorizontalScrollbar;
-  });
 </script>
 
 <section>
   <div class="event-carousel-container">
```

Comment:
This component is no longer a carousel, so it does not feel right to call it that anymore. Something simple like `event-list`?

Reply:
Renamed in <https://github.com/EthanThatOneKid/acmcsuf.com/pull/280/commits/c47f784a8d9e9780bfcd14411fd6d7b2ccee5fe0>.